src/OpenIddict.Client.WebIntegration/OpenIddictClientWebIntegrationHandlers.Userinfo.cs

```diff
@@ -433,10 +433,18 @@ ProviderTypes.Patreon or ProviderTypes.Pipedrive or ProviderTypes.Twitter
                         => new(context.Response["data"]?.GetNamedParameters() ??
                         throw new InvalidOperationException(SR.FormatID0334("data"))),
 
+                    // Linear returns a nested "viewer" object that is itself nested in a GraphQL "data" node.
+                    ProviderTypes.Linear => new(context.Response["data"]?["viewer"]?.GetNamedParameters() ??
```

Comment:
Here we have the potential for a similar issue as for Miro (below) when we query a nested object (as discussed further below for Linear). When we do that, the nested object comes back as a JSON string, e.g.

![image](https://github.com/user-attachments/assets/ac32ae75-999f-4d2a-9775-d3591a88b404)

So, the same issue as below for Miro where we used to resolve this with `ClaimActions.MapJsonSubKey` in the OAuth providers. What is the correct way to handle this here?

Reply:
We can leave it as-is (users will just have to use `JsonObject.Parse(result.Principal.GetClaim("organization"))` to get the organization details) or we can artificially add `organization_id` and `organization_name` nodes (while still returning the raw `organization` JSON node) if you prefer:

```csharp
ProviderTypes.Linear => new(context.Response["data"]?["viewer"]?.GetNamedParameters() ??
    throw new InvalidOperationException(SR.FormatID0334("user")))
{
    ["organization_id"] = context.Response["data"]?["viewer"]?["organization"]?["id"],
    ["organization_name"] = context.Response["data"]?["viewer"]?["organization"]?["name"]
}
```

Reply:
I kept is as-is. The default GraphQL query will not request those organization fields and there is no way for us to know which fields the user will query. Therefore, we cannot assume these fields will be present in the JSON response. Keeping it as-is will keep it in line with the approach we took for Miro (and Basecamp and Harvest)